packages/test-utils/src/wrapper.js

```diff
@@ -24,6 +24,7 @@ export default class Wrapper implements BaseWrapper {
   +options: WrapperOptions
   isFunctionalComponent: boolean
   rootNode: VNode | Element
+  static _instances: ?Array<Wrapper>
 
   constructor(
```

Comment:
**issue:** I think I just found a problem with doing this at the `Wrapper` ctor. Wrappers are created for all components found in the tree on `mount` or `shallowMount`. For instance, when I call `wrapper.find('...')`, it returns another wrapper to me, so I think we'd end up capturing this nested wrapper and double destroying it 😞.

We should probably hook into the `mount` and `shallowMount` functions instead. WDYT?

Reply:
> we'd end up capturing this nested wrapper and double destroying it

@souldzin yes, I think you are right. Even though nothing explodes when destroying twice, it may significantly decrease performance.

> We should probably hook into the mount and shallowMount functions instead.

We would not auto destroy wrappers created via `createWrapper()` or `new Wrapper` then. :thinking:

Reply:
https://github.com/vuejs/vue-test-utils/pull/1248 would allow us to filter out wrappers created by `wrapper.find()` / `wrapper.findAll()`.

Reply:
Maybe we can tell in the ctor if the component is a "root" component or not 🤔... I guess those are the only ones we'd want to automagically destroy.

Reply:
> Maybe we can tell in the ctor if the component is a "root" component 

yes, we can after https://github.com/vuejs/vue-test-utils/pull/1248 because only non-root wrappers would have `wrapper.selector` defined.